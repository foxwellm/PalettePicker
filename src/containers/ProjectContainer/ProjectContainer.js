import React from 'react';
import ProjectCard from '../ProjectCard/ProjectCard';
import { connect } from 'react-redux';

export const ProjectContainer = (props) => {
  const projectCards = props.projects.map(project => {
    const projectPalettes = props.palettes.filter(palette => palette.project_id === project.id);
    return <ProjectCard projectTitle={project.name} palettes={projectPalettes} id={project.id} key={project.id}/>
  });

  return (
    <div>
      <div className='project-container'>
        { projectCards }
      </div>
      <button className='load-more standard-button'>Load More Projects</button>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  projects: state.projects,
  palettes: state.palettes
});

export default connect(mapStateToProps)(ProjectContainer);