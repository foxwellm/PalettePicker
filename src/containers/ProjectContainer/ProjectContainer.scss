.my-masonry-grid {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(400px, 400px));
  margin-top: 40px;
  grid-gap: 20px;
  justify-content: center;
}

.load-more {
  display: block;
  margin: 10px auto;
}

.my-masonry-grid_column {
  background-clip: padding-box;
}

.my-masonry-grid_column > div {
  margin-bottom: 30px;
}