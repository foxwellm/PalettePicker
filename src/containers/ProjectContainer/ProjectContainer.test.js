import React from 'react';
import { ProjectContainer, mapStateToProps } from './ProjectContainer';
import { shallow } from 'enzyme';

describe('ProjectContainer', () => {
  let wrapper;
  let mockProjects;
  let mockPalettes;
  let mockSearching;
  let mockError;

  beforeEach(() => {
    mockError = '';
    mockProjects = [{ id: 1, name: 'Project 1' }, { id: 2, name: 'Project 2' }];
    mockPalettes = [
      {
        id: 1,
        project_id: 1,
        name: 'Palette 1',
        color1: '#ffffff',
        color2: '#ffffff',
        color3: '#ffffff',
        color4: '#ffffff',
        color5: '#ffffff'
      },
      {
        id: 2,
        project_id: 2,
        name: 'Palette 1',
        color1: '#ffffff',
        color2: '#ffffff',
        color3: '#ffffff',
        color4: '#ffffff',
        color5: '#ffffff'
      }
    ];
    mockSearching = false;
    wrapper = shallow(
      <ProjectContainer
        projects={mockProjects}
        palettes={mockPalettes}
        searching={mockSearching}
        error={mockError}
      />
    );
  });

  it('should match the correct snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the correct snapshot when searching is true', () => {
    mockSearching = true;
    wrapper = shallow(
      <ProjectContainer
        projects={mockProjects}
        palettes={mockPalettes}
        searching={mockSearching}
        error={mockError}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return an object of props', () => {
      const mockState = {
        projects: mockProjects,
        palettes: mockPalettes,
        searching: mockSearching,
        error: mockError,
        otherState: false,
      };
      const expected = {
        projects: mockProjects,
        palettes: mockPalettes,
        searching: mockSearching, 
        error: mockError,
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

});