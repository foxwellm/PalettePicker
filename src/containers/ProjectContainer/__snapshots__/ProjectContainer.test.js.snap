// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectContainer should match the correct snapshot 1`] = `
<Masonry
  breakpointCols={
    Object {
      "1100": 3,
      "500": 1,
      "700": 2,
      "default": 3,
    }
  }
  className="my-masonry-grid"
  columnClassName="my-masonry-grid_column"
>
  <Connect(ProjectCard)
    id={1}
    key="1"
    palettes={
      Array [
        Object {
          "color1": "#ffffff",
          "color2": "#ffffff",
          "color3": "#ffffff",
          "color4": "#ffffff",
          "color5": "#ffffff",
          "id": 1,
          "name": "Palette 1",
          "project_id": 1,
        },
      ]
    }
    projectTitle="Project 1"
  />
  <Connect(ProjectCard)
    id={2}
    key="2"
    palettes={
      Array [
        Object {
          "color1": "#ffffff",
          "color2": "#ffffff",
          "color3": "#ffffff",
          "color4": "#ffffff",
          "color5": "#ffffff",
          "id": 2,
          "name": "Palette 1",
          "project_id": 2,
        },
      ]
    }
    projectTitle="Project 2"
  />
</Masonry>
`;

exports[`ProjectContainer should match the correct snapshot when searching is true 1`] = `
<Masonry
  breakpointCols={
    Object {
      "1100": 3,
      "500": 1,
      "700": 2,
      "default": 3,
    }
  }
  className="my-masonry-grid"
  columnClassName="my-masonry-grid_column"
>
  <Connect(ProjectCard)
    id={1}
    key="1"
    palettes={
      Array [
        Object {
          "color1": "#ffffff",
          "color2": "#ffffff",
          "color3": "#ffffff",
          "color4": "#ffffff",
          "color5": "#ffffff",
          "id": 1,
          "name": "Palette 1",
          "project_id": 1,
        },
      ]
    }
    projectTitle="Project 1"
  />
  <Connect(ProjectCard)
    id={2}
    key="2"
    palettes={
      Array [
        Object {
          "color1": "#ffffff",
          "color2": "#ffffff",
          "color3": "#ffffff",
          "color4": "#ffffff",
          "color5": "#ffffff",
          "id": 2,
          "name": "Palette 1",
          "project_id": 2,
        },
      ]
    }
    projectTitle="Project 2"
  />
</Masonry>
`;
